internal/command/machine/clone.go

```diff
@@ -244,14 +244,19 @@ func runMachineClone(ctx context.Context) (err error) {
 				fmt.Fprintf(io.Out, "Creating new volume from snapshot: %s\n", colorize.Bold(*snapshotID))
 			}
 
+			var hostDedicationID string
+			if vol != nil {
+				hostDedicationID = vol.HostDedicationID
+			}
+
 			volInput := api.CreateVolumeRequest{
 				Name:              mnt.Name,
 				Region:            region,
 				SizeGb:            &mnt.SizeGb,
 				Encrypted:         &mnt.Encrypted,
 				SnapshotID:        snapshotID,
 				RequireUniqueZone: api.Pointer(flag.GetBool(ctx, "volume-requires-unique-zone")),
-				HostDedicationId:  vol.HostDedicationID,
+				HostDedicationId:  hostDedicationID,
```

Comment:
Perhaps this should be set to the host dedication ID of the machine being cloned? (I'm not up to speed on host dedication though, so feel free to disregard!)